python/cugraph-dgl/cugraph_dgl/nn/conv/base.py

```diff
@@ -48,3 +49,85 @@ def pad_offsets(self, offsets: torch.Tensor, size: int) -> torch.Tensor:
         self._cached_offsets_fg[offsets.numel() : size] = offsets[-1]
 
         return self._cached_offsets_fg[:size]
+
+
+class SparseGraph(object):
+    r"""A god-class to store different sparse formats needed by cugraph-ops
+    and facilitate sparse format conversions.
+
+    Parameters
+    ----------
+    size: tuple of int
+        Size of the adjacency matrix: (num_src_nodes, num_dst_nodes).
+
+    src_ids: torch.Tensor
+        Source indices of the edges.
+
+    dst_ids: torch.Tensor, optional
+        Destination indices of the edges.
+
+    csrc_ids: torch.Tensor, optional
+        Compressed source indices. It is a monotonically increasing array of
+        size (num_src_nodes + 1,). For the k-th source node, its neighborhood
+        consists of the destinations between `dst_indices[csrc_indices[k]]` and
+        `dst_indices[csrc_indices[k+1]]`.
```

Comment:
I have a question regarding when `num_dst_nodes` > `len(cdst_ids)-1` for this case. 

Lets look at below case: 

```python3
cdst_ids (Compressed Destinations Indices):0,2,5,7
```
```python3
src_indices: 1,2,2,3,4,4,5
```
I believe following will work (please correct me):

```python3
num_src_nodes = 6
num_dst_nodes = 3 
```

And i guess below will fail ((please correct me):

```python3
num_src_nodes = 6
num_dst_nodes = 5 # Modified it to a higher value to ensure alignment for output nodes that are missing
```

**Question:** 
So this will have to handled by ensuring correct creation because we want to handle alignment problem b/w blocks.

Reply:
It should be illegal when `num_dst_nodes` != `len(cdst_ids)-1`. I will improve the error handling in this case. For example, pyg does lots of assertions to check the size. We should throw proper exceptions.

```
cdst_ids (Compressed Destinations Indices):0,2,5,7
src_indices: 1,2,2,3,4,4,5
```
In your example with num_src_nodes = 6, num_dst_nodes = 3, this translates to a COO of
(1,2,2,3,4,4,5)
(0,0,1,1,1,2,2)

With num_src_nodes = 6, num_dst_nodes = 5, the constructor should have failed, unless `cdst_ids` is augmented (cdst_ids = 0,2,5,7,7,7).

Reply:
Thanks. Yup, this is what i was expecting. We will just make sure that the changes @seunghwak  from cugraph sampling ensures that all the MFGs line up.